services/cache.ts

```diff
@@ -0,0 +1,70 @@
+interface CacheValue {
+  value: string;
+  updatedAt: number;
+}
+
+class Cache {
+  cache: Map<string, CacheValue>;
+  maxLength: number;
+
+  constructor() {
+    this.cache = new Map();
+    this.maxLength = 100;
+  }
+
+  has(key: string) {
+    return this.cache.has(key);
+  }
+
+  set(key: string, value: string) {
+    if (this.length() > this.maxLength) {
+      const leastUsed = this.leastRecentlyUsed();
+      this.delete(leastUsed);
+    }
+
+    return this.cache.set(key, {
+      value,
+      updatedAt: Date.now(),
+    });
+  }
+
+  getValue(key: string): string {
+    if (!this.has(key)) {
+      return '';
+    }
+
+    return this.cache.get(key).value;
+  }
+
+  delete(key: string) {
+    return this.cache.delete(key);
+  }
+
+  clear() {
+    return this.cache.clear();
+  }
+
+  length() {
+    return Array.from(this.cache.keys()).length;
+  }
+
+  leastRecentlyUsed() {
```

Comment:
It seems the updatedAt is not being reset if we grab the avatar. This function name implies that the updatedAt will be updated every time you re-grab the avatar.